myuw-ios/AppAuthTest.swift

```diff
@@ -0,0 +1,452 @@
+            return
+        }
+
+        let request: OIDRegistrationRequest = OIDRegistrationRequest(configuration: configuration,
+                                                                     redirectURIs: [redirectURI],
+                                                                     responseTypes: nil,
+                                                                     grantTypes: nil,
+                                                                     subjectType: nil,
+                                                                     tokenEndpointAuthMethod: "client_secret_post",
+                                                                     additionalParameters: nil)
+
+        // performs registration request
+        print("Initiating registration request")
+
+        OIDAuthorizationService.perform(request) { response, error in
+
+            if let regResponse = response {
+                self.setAuthState(OIDAuthState(registrationResponse: regResponse))
+                print("Got registration response: \(regResponse)")
+                callback(configuration, regResponse)
+            } else {
+                print("Registration error: \(error?.localizedDescription ?? "DEFAULT_ERROR")")
+                self.setAuthState(nil)
+            }
+        }
+    }
+    
+    func doAuthWithAutoCodeExchange(configuration: OIDServiceConfiguration, clientID: String, clientSecret: String?) {
+        
+        print("doAuthWithAutoCodeExchange")
+        
+        guard let redirectURI = URL(string: kRedirectURI) else {
+            print("Error creating URL for : \(kRedirectURI)")
+            return
+        }
+        
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
+            print("Error accessing AppDelegate")
+            return
+        }
+
+        // builds authentication request
+        let request = OIDAuthorizationRequest(configuration: configuration,
+                                              clientId: clientID,
+                                              clientSecret: clientSecret,
+                                              scopes: [OIDScopeOpenID, OIDScopeProfile, OIDScopeEmail],
+                                              redirectURL: redirectURI,
+                                              responseType: OIDResponseTypeCode,
+                                              additionalParameters: nil)
+
+        // performs authentication request
+        print("Initiating authorization request with scope: \(request.scope ?? "DEFAULT_SCOPE")")
+
+        appDelegate.currentAuthorizationFlow = OIDAuthState.authState(byPresenting: request, presenting: self) { authState, error in
+            if let authState = authState {
+                self.setAuthState(authState)
+                print("Got authorization tokens. Access token: \(authState.lastTokenResponse?.accessToken ?? "DEFAULT_TOKEN")")
```

Comment:
Rather than print, we should start logging these with an appropriate loglevel.

Reply:
using os_log now.